hermes-metrics/src/main/java/pl/allegro/tech/hermes/metrics/HermesCounterWithRate.java

```diff
@@ -0,0 +1,4 @@
+package pl.allegro.tech.hermes.metrics;
+
+public interface HermesCounterWithRate extends HermesCounter, HermesRateMeter {
```

Comment:
Do we really need to have a separate interface? AFAIK, Dropwizard's Meter also provides `count`. The only place where we benefit from having this distinction is the `published` counter, as we don't produce m1_rate for it. So, perhaps we can provide a dedicated implementation for this case (even as an anonymous class).

Reply:
We can drop the interface by directly exposing concrete implementations of `HermesCounters`: `DefaultHermesCounter`, `MeterBackedHermesCounter` from `HermesCounter` - please see if thats what you expected: https://github.com/allegro/hermes/pull/1695/commits/ce935d08596e1a14768134f44534a6a5a08a87ec I don't think its possible to provide the implementation as anonymous class, can you perhaps write a snippet of what you had in mind?

Reply:
My idea was as follows: 

Let's have the `HermesCounter` interface: 

```java
public interface HermesCounter {
   
    void increment(long size);

    default void increment() {
        increment(1L);
    }
    
    double getOneMinuteRate();
}
```

and two classes implementing it (forget about my suggestion regarding the anonymous class):

```java
public class DefaultHermesCounterWithRate implements HermesCounter {
    private final io.micrometer.core.instrument.Counter micrometerCounter;
    private final com.codahale.metrics.Meter graphiteMeter;

    DefaultHermesCounterWithRate(io.micrometer.core.instrument.Counter micrometerCounter,
                                    com.codahale.metrics.Meter graphiteMeter) {
        this.micrometerCounter = micrometerCounter;
        this.graphiteMeter = graphiteMeter;
    }
    
    ...

    public void increment(long size) {
        this.micrometerCounter.increment((double) size);
        this.graphiteMeter.mark(size);
    }

    @Override
    public double getOneMinuteRate() {
        return graphiteMeter.getOneMinuteRate();
    }
}
```

```java
public class DefaultHermesCounter implements HermesCounter {
    private final io.micrometer.core.instrument.Counter micrometerCounter;
    private final com.codahale.metrics.Counter graphiteCounter;

    DefaultHermesCounter(io.micrometer.core.instrument.Counter micrometerCounter,
                                   com.codahale.metrics.Counter graphiteCounter) {
        this.micrometerCounter = micrometerCounter;
        this.graphiteCounter = graphiteCounter;
    }
   
    ...

    public void increment(long size) {
        this.micrometerCounter.increment((double) size);
        this.graphiteCounter.inc(size);
    }

    @Override
    public double getOneMinuteRate() {
        // I forgot about this method :(. Can we get the rate from the micrometer counter? We will need to do that in the future either way.
    }
}
```

As you see, my idea has one flaw (see the comment in the `DefaultHermesCounter#getOneMinuteRate` method). Considering that we will get rid of `HermesCounterWithRate` after the migration (is my assumption correct?), there is no point in providing a fake implementation of `getOneMinuteRate` in `DefaultHermesCounter`.

Reply:
I don't think there is an easy way to extract `m1_rate` client side from micrometer (I couldn't find a built-in option to do that). So lets assume that calculating `oneMinuteRate` based on `micrometer#Counter` requires querying metric backend or some fancy client side aggregation and is out of scope of this PR. 

What in that case? _Implementing_ `getOneMinuteRate` in `DefaultHermesCounter` with something like 

```java
@Override
    public double getOneMinuteRate() {
        throw UnimplementedException()
    }
```

seems dangerous as we loose compile-time safety and someone _could_ use it by mistake. That's why I separated `HermesCounter` and `HermesCounterWithRate` originally. What do you think?

Reply:
By `fake implementation` I meant something like: 

```java
    @Override
    public double getOneMinuteRate() {
        return 0d;
    }
```

or as you suggested: 
```java
    @Override
    public double getOneMinuteRate() {
        throw new UnimplementedException()
    }
```

> I don't think there is an easy way to extract m1_rate client side from micrometer 

So, how are we going to implement `getOneMinuteRate` after removing dropwizard?

Reply:
> So, how are we going to implement getOneMinuteRate after removing dropwizard? 

Perhaps @faderskd had an idea as he originally introduced implementations of `HermesRateMeter` which were solely based on dropwizard.  I'd say the options are:
1. querying the metric backend periodically
2. implementing rate metering on application side

Anyway, I thought this was purposefully left to deal with later on so I didn't want to do it in this PR

Reply:
OK, let's leave it as it is now.